InnerEye/ML/Histopathology/datamodules/base_module.py

```diff
@@ -46,16 +50,25 @@ def __init__(self, root_path: Path, max_bag_size: int = 0, batch_size: int = 1,
         :param cache_mode: The type of caching to perform, i.e. whether the results of all
         transforms up to the first randomised one should be computed only once and reused in
         subsequent iterations:
-          - `MEMORY`: the entire transformed dataset is kept in memory for fastest access;
-          - `DISK`: each transformed sample is saved to disk and loaded on-demand;
-          - `NONE` (default): no caching is performed.
-        :param save_precache: Whether to pre-cache the entire transformed dataset upfront and save
+          - `MEMORY`: MONAI CacheDataset is used, the entire transformed dataset is kept in memory for fastest access;
+          - `DISK`: MONAI PersistentDataset is used, each transformed sample is saved to disk and loaded on-demand;
+          - `NONE` (default): standard MONAI dataset is used, no caching is performed.
+        :param precache_location: Whether to pre-cache the entire transformed dataset upfront and save
         it to disk. This is done once in `prepare_data()` only on the local rank-0 process, so
-        multiple processes can afterwards access the same cache without contention in DDP settings.
+        multiple processes can afterwards access the same cache without contention in DDP settings. 3 options are
+        available:
+          - `CPU`: each transformed sample is saved to disk and loaded into CPU memory on-demand;
+          - `GPU`: each transformed sample is saved to disk and loaded into GPU memory on-demand;
```

Comment:
Given that this argument applies to `MEMORY` mode only, consider rephrasing these descriptions to clarify that the whole dataset is saved and loaded at once, rather than on-demand.

Reply:
done